azuread/resource_domain_service.go

```diff
@@ -0,0 +1,492 @@
+			return fmt.Errorf("Error setting `notification_settings`: %+v", err)
+		}
+		d.Set("subnet_id", domainServiceProperties.SubnetID)
+	}
+
+	return nil
+}
+
+func resourceArmDomainServiceUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).domainServicesClient
+	ctx := meta.(*ArmClient).StopContext
+
+	name := d.Get("name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+	domainSecuritySettings := d.Get("domain_security_settings").([]interface{})
+	filteredSync := d.Get("filtered_sync").(string)
+	ldapsSettings := d.Get("ldaps_settings").([]interface{})
+	notificationSettings := d.Get("notification_settings").([]interface{})
+	subnetId := d.Get("subnet_id").(string)
+
+	domainService := aad.DomainService{
+		DomainServiceProperties: &aad.DomainServiceProperties{
+			DomainName:             &name,
+			DomainSecuritySettings: expandArmDomainServiceDomainSecuritySettings(domainSecuritySettings),
+			FilteredSync:           aad.FilteredSync(filteredSync),
+			LdapsSettings:          expandArmDomainServiceLdapsSettings(ldapsSettings),
+			NotificationSettings:   expandArmDomainServiceNotificationSettings(notificationSettings),
+			SubnetID:               &subnetId,
+		},
+	}
+
+	future, err := client.Update(ctx, resourceGroup, name, domainService)
+	if err != nil {
+		return fmt.Errorf("Error updating Domain Service %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("Error waiting for update of Domain Service %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	return resourceArmDomainServiceRead(d, meta)
+}
+
+func resourceArmDomainServiceDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).domainServicesClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := azure.ParseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	name := id.Path["domainServices"]
+
+	future, err := client.Delete(ctx, resourceGroup, name)
+	if err != nil {
+		if response.WasNotFound(future.Response()) {
+			return nil
+		}
+		return fmt.Errorf("Error deleting Domain Service %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		if !response.WasNotFound(future.Response()) {
+			return fmt.Errorf("Error waiting for deleting Domain Service %q (Resource Group %q): %+v", name, resourceGroup, err)
+		}
+	}
+
+	return nil
+}
+
+func domainServiceControllerRefreshFunc(ctx context.Context, client *aad.DomainServicesClient, resourceGroup, name string) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		log.Printf("waiting for both Domain Service Controller up...")
+		resp, err := client.Get(ctx, resourceGroup, name)
+		if err != nil {
+			return nil, "pending", err
```

Comment:
I think this should be
```suggestion
			return nil, "error", err
```